app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt

```diff
@@ -0,0 +1,235 @@
+import org.oppia.android.testing.junit.InitializeDefaultLocaleRule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.AssetModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Singleton
+
+/** Tests for [DrawableBindingAdaptersTest]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = DrawableBindingAdaptersTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class DrawableBindingAdaptersTest {
+
+  // TODO(#3059): Add more tests for other BindableAdapters present in [ImageViewBindingAdapters].
+
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @get:Rule
+  var activityRule: ActivityScenarioRule<DrawableBindingAdaptersTestActivity> =
+    ActivityScenarioRule(
+      Intent(
+        ApplicationProvider.getApplicationContext(),
+        DrawableBindingAdaptersTestActivity::class.java
+      )
+    )
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    Intents.init()
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  // from a color int
+  private var colorRgb: Int = Color.valueOf(-0x10000).toArgb()
+
+  @Test
+  fun testSetBackgroundColor_hasCorrectBackgroundColor() {
+    activityRule.scenario.onActivity {
+      val view: View = getView(it)
+      setBackgroundColor(view, /* colorRgb= */ colorRgb)
+      assertThat((view.background as ColorDrawable).color).isEqualTo(colorRgb)
+    }
+  }
+
+  @Test
+  fun testSetBackgroundDrawable_hasCorrectBackgroundDrawable() {
+    activityRule.scenario.onActivity {
+      val view: View = getView(it)
+      setBackgroundDrawable(view, /* colorRgb= */ colorRgb)
+      assertThat((view.background as GradientDrawable).color?.defaultColor).isEqualTo(colorRgb)
+    }
+  }
+
+  @Test
+  fun testSetTopBackgroundDrawable_hasCorrectTopBackgroundDrawable() {
+    activityRule.scenario.onActivity {
+      val view: View = getView(it)
+      setTopBackgroundDrawable(view, /* colorRgb= */ colorRgb)
+      assertThat((view.background as GradientDrawable).color?.defaultColor).isEqualTo(colorRgb)
```

Comment:
Shouldn't these be setting actually drawable IDs, not colors? They seem incorrect as-is.

Reply:
Agreed

Reply:
While comparing, Drawable was not working out, I was facing the same issue as what we faced during the meeting while writing the tests for #4060. I was able to get the Drawable object for the view, but they both were different Drawable objects, so I had to compare colors instead of Drawable. Also, an Int color cannot be typecast to a `GradientDrawable`,

![image](https://user-images.githubusercontent.com/78549195/149899440-cf71a9cb-4299-4405-a3df-1aae000676eb.png)

As of now, I was only able to find this as a solution to compare default colors for Gradient Drawable to Input color. 

One solution I have in my mind is we can get another view and make the same changes to what we are doing originally, and we can get the `view.background` as Gradient Drawable for both, and then we can compare and that should work. I will try implementing and update you on that.

If you guys have another approach, please let me know.

Reply:
This is much clearer. I also misunderstood one part: I thought a color res ID was supposed to be passed into the adapter method, not a color value. The latest changes LGTM--thanks.